backend/src/services/order/saveOrder.ts

```diff
@@ -122,27 +123,20 @@ export const saveOrder = async (
   const productCategories = await AppDataSource.getRepository(
     ProductCategory,
   ).find({ relations: { products: true } });
-  if (order) {
-    order.offer = body.offer;
-    order.depotId = body.depotId;
-    order.alternateDepotId = body.alternateDepotId;
-    order.productConfiguration = JSON.stringify(productCategories);
-    order.offerReason = body.offerReason || "";
-    order.category = body.category;
-    order.categoryReason = body.categoryReason || "";
-    await AppDataSource.getRepository(Order).save(order);
-  } else {
+
+  if (!order) {
     order = new Order();
     order.userId = requestUserId;
-    order.offer = body.offer;
-    order.offerReason = body.offerReason || "";
-    order.categoryReason = body.categoryReason || "";
-    order.alternateDepotId = body.alternateDepotId;
-    order.depotId = body.depotId;
-    order.category = body.category;
-    order.productConfiguration = JSON.stringify(productCategories);
-    await AppDataSource.getRepository(Order).save(order);
   }
+  order.offer = body.offer;
+  order.depotId = body.depotId;
+  order.alternateDepotId = body.alternateDepotId;
+  order.productConfiguration = JSON.stringify(productCategories);
```

Comment:
@andreashille: I couldn't find where `Order.productConfiguration` is acutally used and it seems a bit much to store a JSON representation of the whole product configuration with each order.

Am I missing something or is this obsolete?

Reply:
Security reasons. We save the configuration to know when a admin changed the products for instance prices or so to be able to check the configuration at time of saving in case of a disagreement